### PR TITLE
[ci, model] test: add bitwise logits-equal tests for v5 models

### PR DESCRIPTION
**Pull request by TimYangst**
### What does this PR do?

> Adds a v5 sibling to `tests/models/test_models_logits_equal.py` (the v4 test). For every patchgen-migrated model, builds a pristine HuggingFace v5 model and the matching VeOmni `generated/patched_modeling_*_gpu.py` model side-by-side, copies HF weights, runs forward, and asserts bitwise-equal logits. This catches drift between the two whenever a patchgen config or kernel-dispatch path changes.
>
> To get there, also migrates the last few models that still did unconditional `class_replacement` swaps (always-on Liger) to the OpSlot-guard pattern that the rest already use. With OpSlot guards, `rms_norm_implementation="eager"` actually falls through to the HF reference path, which is what makes the bitwise test possible.

### Checklist Before Starting

- Search for relative PRs/issues and link here: companion to #721 (v4 logits-equal test); follows #714 (Qwen3.5 OpSlot dispatch).
- PR title follows `[{modules}] {type}: {description}` format

### Test

> 18/18 cases bitwise-equal on a single A100 in ~17s under transformers 5.2.0. Each model gets eager+fp32 (RNG-init parity baseline) plus FA2/SDPA+bf16 (the dtype real users hit):
>
> | Family | Cases |
> | ------ | ----- |
> | Causal-LM | qwen2, qwen3, qwen3_moe |
> | Qwen3.5 text-only sub-config | qwen3_5, qwen3_5_moe |
> | VLM full forward (image + text) | qwen2_vl, qwen2_5_vl, qwen3_vl, qwen3_vl_moe |
> | Omni thinker forward | qwen3_omni_moe |
>
> Existing forward+backward parity test still passes after the migrations: `pytest tests/models/test_models_patch.py -k qwen3_5` ✓.
>
> CI: wired into the `[transformers-v5]` step of `.github/workflows/gpu_unit_tests.yml`. `make patchgen` drift check + `make quality` clean.

### API and Usage Example

> N/A — internal test + patch-config refactor only.
>
> User-visible default change: qwen2 / qwen3_5 / qwen3_moe GPU loading no longer applies LigerKernel RMSNorm/MLP automatically. The default is the eager HF code; LigerKernel is opt-in via `OpsImplementationConfig(rms_norm_implementation="liger_kernel", swiglu_mlp_implementation="liger_kernel", rotary_pos_emb_implementation="liger_kernel")`. This matches the qwen3 / qwen3_5_moe / qwen3_vl_moe / qwen3_omni_moe semantics that already shipped this way.

### Design & Code Changes

> #### Test (`tests/models/test_models_logits_equal_v5.py`)
>
> - Single parametrized test over all migrated models. A `Case.kind` field selects between four flows: causal-LM, qwen3_5 text-only sub-config (extracts `text_config` and forces all-`full_attention` to dodge the fla-vs-conv1d kernel divergence), VLM full forward (single dummy 2x2 image), Omni thinker forward.
> - No `hf_unpatch.py` analogue needed — v5 patchgen files don't mutate the `transformers.*` namespace, so HF and VeOmni can be built side-by-side.
> - MoE cases force `_experts_implementation="eager"` on the HF side. HF's default `"grouped_mm"` (`torch._grouped_mm`) crashes or numerically diverges from VeOmni's eager per-expert loop on the toy configs.
> - qwen3_5_moe falls back to SDPA on bf16: the toy's `(num_kv_heads=2, head_dim=256, seq_len=32)` shape crashes `flash_attn_varlen_forward` on both HF and VeOmni.
> - Module-scoped 1-rank NCCL fixture so VeOmni's SP-aware FA wrappers see a default group at `sp_size=1`.
>
> #### Migrations to OpSlot guard (`*_gpu_patch_gen_config.py`)
>
> Same shape on every migrated model: drop `create_patch_from_external` / `replace_function`, declare an `OpSlot(op, variant)` in the post-import block, and `override_method` the relevant `forward` with a top-of-function `if veomni_<op>.use_non_eager_impl: return veomni_<op>(...)` guard followed by the original HF code.
>
> | Model | Migrated this PR |
> | ----- | ---------------- |
> | qwen3_5 | RMSNorm |
> | qwen2 | RMSNorm + MLP + RoPE |
> | qwen3_moe | RMSNorm + MLP + RoPE |
>
> Plus a small qwen3_moe router fix: `Qwen3MoeTopKRouter.forward` was casting post-softmax top-k weights down to `input_dtype` (bf16). HF keeps them in fp32 (the softmax dtype). Under FA2+bf16 this gave ~1.7e-1 logits drift on the 1-token toy. Cast to `routing_weights.dtype` (fp32) instead — no-op on the fused MoE path which casts itself at the call site.
>
> NPU patch configs are untouched; they already used `override_method` (no class swaps), and adding OpSlot guards to NPU is a separate follow-up.

### Coverage gap (eventual goal: every v5 model)

> Patchgen-migrated models not yet in this test:
>
> - **glm_moe_dsa** — has v5 patchgen but no toy config under `tests/toy_config/`. Add the toy config + a case here to cover.
>
> Models still on the v4 monkey-patch path (no v5 patchgen yet):
>
> - **Standard LMs / VLMs / Omni**: deepseek_v3, llama, seed_oss, qwen2_5_omni. Each needs a v5 patch config + regenerated modeling, and a case added here once that lands.
> - **Diffusion / vision-only**: janus, flux, movqgan, wan. These don't produce token logits and may need a different parity test (e.g. compare encoder/decoder hidden states); out of scope for this file.
>
> CI step in `gpu_unit_tests.yml` is already wired, so adding a case in this file is the only test-side work for each future model.

### Checklist Before Submitting

- [ ] Read the [Contribute Guide](https://github.com/ByteDance-Seed/VeOmni/blob/main/CONTRIBUTING.md)
- [ ] Applied pre-commit checks
- [ ] Added/updated documentation
- [ ] Added tests to CI workflow (or explained why not feasible)

### CODE REVIEW

**Review by Luosuu — APPROVED** [review]

File: tests/models/test_models_logits_equal_v5.py [hunk 1]
```diff
@@ -0,0 +1,349 @@
...
+    """Text-only input_ids and forward kwargs.
+
+    Vocab floor 32000 dodges multimodal placeholder ids. ``cu_seq_lens_q``
+    is required by VeOmni's patched ``Qwen3_5DecoderLayer.forward``; HF's
+    decoder ignores it via ``**kwargs``.
+    """
+    gen = torch.Generator(device=device).manual_seed(0)
+    vocab = max(min(text_config.vocab_size, 200000), 32000)
+    seq_len = 32
+    input_ids = torch.randint(32000, vocab, (1, seq_len), device=device, dtype=torch.long, generator=gen)
```
[the remark this answers is not in the document]
Comment (TimYangst) [reply tests/models/test_models_logits_equal_v5.py]: Outdated
